common4j/src/main/com/microsoft/identity/common/java/opentelemetry/FidoTelemetryHelper.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.opentelemetry;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+
+/**
+ * Main implementation of IFidoTelemetryHelper.
+ */
+public class FidoTelemetryHelper implements IFidoTelemetryHelper {
```

Comment:
Instead of a helper class, why not directly populate span (current span or new span per your case) in the at the place you would call these methods. Common attributes can be emitted in parent abstract class?

Reply:
I could do that. This feature relies on a lot of abstraction since it pulls in the credman api and potentially others later, so I got into the habit of creating helpers when using other apis. But given my current plan is to only emit telemetry in one class (the ChallengeHandler class), there isn't a need here for a helper.

Reply:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2183/commits/7aedde14b813a93bf9a198865da859b68af12806